ci-automation/vendor-testing/vmware.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+# Copyright (c) 2022 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+# Test execution script for the VMware ESX vendor image.
+# This script is supposed to run in the SDK container.
+
+work_dir="$1"; shift
+arch="$1"; shift
+vernum="$1"; shift
+tapfile="$1"; shift
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/ci_automation_common.sh
+source sdk_lib/sdk_container_common.sh
+
+mkdir -p "${work_dir}"
+cd "${work_dir}"
+channel="$(get_git_channel)"
+if [[ "${channel}" = 'developer' ]]; then
+    channel='alpha'
+fi
+testscript="$(basename "$0")"
+
+# We never ran VMware ESX on arm64, so for now fail it as an
+# unsupported option.
+if [[ "${arch}" == "arm64" ]]; then
+    echo "1..1" > "${tapfile}"
+    echo "not ok - all qemu tests" >> "${tapfile}"
+    echo "  ---" >> "${tapfile}"
+    echo "  ERROR: ARM64 tests not supported on VMware ESX." | tee -a "${tapfile}"
+    echo "  ..." >> "${tapfile}"
+    exit 1
+fi
+
+# Fetch image if not present.
+if [ -f "${ESX_IMAGE_NAME}" ] ; then
+    echo "++++ ${testscript}: Using existing ${work_dir}/${ESX_IMAGE_NAME} for testing ${vernum} (${arch}) ++++"
+else
+    echo "++++ ${testscript}: downloading ${ESX_IMAGE_NAME} for ${vernum} (${arch}) ++++"
+    copy_from_buildcache "images/${arch}/${vernum}/${ESX_IMAGE_NAME}" .
+fi
+
+set -x
+
+# krnowak: This is what the old script in jenkins did. I'm not sure we
+# should be backporting this - it seems to be very specific for our
+# setup.
+#
+# # Delete every VM that is running because we'll use all available spots
+# ore esx --esx-config-file "${ESX_CONFIG_FILE}" remove-vms || :
```

Comment:
The static IP allocation inside kola assumes exclusive usage when `StaticIPs != 0`. We can guard this command here by a check of the json file's value with `jq`.

Reply:
How does that play with the already running tests in CI when someone wants to reproduce something by running the tests locally?

Reply:
If you want to use it locally, you should also "reserve" the lock under http://192.168.42.7:8080/lockable-resources/ or your VMs will compete with the CI VMs over the same IP address…

The removal ensures that the CI isn't entering a broken state if there are left over VMs from manual usage or a forcefully canceled Jenkins job.

Reply:
This is done.